scapy/modules/packet_viewer/button_bar.py

```diff
@@ -0,0 +1,141 @@
+
+    def keypress(self, size, key):
+        # type: (int, str) -> None
+        """
+        Handle editing keystrokes, return None to not forward key press.
+        :param size:
+        :param key: Name of key pressed.
+        """
+        if key in self._actions:
+            self._execute_and_change_state(key)
+
+    def _execute_and_change_state(self, key):
+        # type: (str) -> None
+        """
+        Executes action for a key and updates the according button text
+        :param key: Key to execute
+        """
+        action = self._actions[key]
+        action.execute()
+
+        btn = self._key_button_map[key]
+        btn.set_label(("row_focused", action.text))
+
+    # noinspection PyProtectedMember
+    def _create_button(self, cmd):
+        # type: (Tuple[str, Action]) -> Button
+        """
+        Helper function to create a Button object for a command
+        :param cmd: Tuple of key and Action object
+        :return: Button for this Action
+        """
+        key, action = cmd
+
+        btn = Button(("row_focused", action.text),
+                     on_press=self._execute_and_change_state,
+                     user_data=key)
+        # We need to access the underlying Columns widget
+        cols = btn._w
+        # We don't want any dividing chars
+        cols.dividechars = 0
+        # Set the prefix and make it pack instead of "<" and fixed length
+        cols.contents[0] = (Text(key.upper()), cols.options("pack"))
+        # Remove the ">" behind the actual button text
+        del cols.contents[2]
+        # len(text) + 1 hides the cursor
+        cols.contents[1][0]._cursor_position = len(btn.label) + 1
+        # Ensure buttons won't gain focus but they are still clickable
+        cols._selectable = False
+        return btn
+
+
+class Action(object):
+    """
+    Helper class to store a list of texts and functions. On every execute,
+    the internal index increases. The internal index points to the current
+    text and function. If the index points to the last function, the next
+    execution causes a roll-over to index zero.
+    """
+    def __init__(self, texts, funcs, state_index=0):
+        # type: (List[str], List[Callable[[], None]], int) -> None  # noqa: E501
+        """
+        Initialize an Action object
+        :param texts: A list of texts. Has to have the same order as funcs.
+        :param funcs: A list of functions. Has to have the same order as texts.
+        :param state_index: initial index if necessary
+        """
+        self._texts = texts
+        self._funcs = funcs
+        self._state_index = state_index
+        if len(self._texts) != len(self._funcs):
+            raise AssertionError("The lists texts and funcs need to have "
+                                 "the same length")
+        if self._state_index > len(self._texts):
+            raise AssertionError("State index can't be greater than length "
```

Comment:
`state_index` is just a name at this point. I raise the error after a value doesn't match my requirements, therefor it should be a AssertionError in my opinion. I don't access an index in a container, which would cause an IndexError